.changeset/polite-schools-burn.md

```diff
@@ -0,0 +1,7 @@
+---
+'@shopify/cli-kit': patch
+'@shopify/theme': patch
+---
+
+- Introduce REST Admin API client on `@shopify/cli-kit`
+- Improve the DX of the `shopify theme open` command, by adopting the CLI UI kit
```

Comment:
All these changes feel more like a "minor" than a patch, what do you think?

Reply:
@isaacroldan This PR have many additions to provide a more polished DX. Still, this PR is also not introducing any feature or functional change in the commands, so, from the users perspective, I'd still consider it a "minor" change.